crypto/keyring/keyring.go

```diff
@@ -848,146 +847,100 @@ func (ks keystore) newRecord(name string, pk types.PubKey, item isRecord_Item) (
+			return migrated, fmt.Errorf(" not a keyring.ErrKeyNotFound, err: %s", err)
 		}
 		// key not found, all good: assume version = 0
 	} else {
 		if len(item.Data) != 4 {
-			return false, sdkerrors.ErrInvalidVersion.Wrapf(
+			return migrated, sdkerrors.ErrInvalidVersion.Wrapf(
 				"Can't migrate the keyring - the stored version is malformed: [%s]: %s",
 				item.Description, string(item.Data))
 		}
 		version = binary.LittleEndian.Uint32(item.Data)
 	}
-	return ks.migrate(version, item)
-}
-
-func (ks keystore) migrate(version uint32, i keyring.Item) (bool, error) {
+	
 	if version == CURRENT_VERSION {
 		// return nil
-		return false, fmt.Errorf("versions match")
+		return migrated, fmt.Errorf("versions match")
 	}
 	if version > CURRENT_VERSION {
-		return false, sdkerrors.ErrInvalidVersion.Wrapf(
+		return migrated, sdkerrors.ErrInvalidVersion.Wrapf(
 			"Can't migrate the keyring - wrong keyring version: [%v]: %v, expected version to be max %d",
-			i.Description, string(i.Data), CURRENT_VERSION)
-	}
-	
-	// TODO i can use map or array of boleeans to check if no error occured thenw eu pdate the version otherwise
-	// return nil or error
-	// 6. at the end of the loop update version
-	migrated, err := ks.performMigration()
-	if !migrated {
-		return migrated, err
-	} else {
-		if err := ks.setVersion(); err != nil {
-			return migrated, err
-		}
+			item.Description, string(item.Data), CURRENT_VERSION)
 	}
 	
-	return migrated, nil
-}
-
-func (ks keystore) performMigration() (bool, error) {
 	keys, err := ks.db.Keys()
 	if err != nil {
 		// return err
-		return false, fmt.Errorf("Keys() error, err: %s", err)
+		return migrated, fmt.Errorf("Keys() error, err: %s", err)
 	}
 
-	migrations := make([]bool, len(keys))
-
-	for i, key := range keys {
+	for _, key := range keys {
 		if !strings.HasSuffix(key, infoSuffix) {
 			fmt.Printf("key %s has no infoSuffix", key)
 			continue
 		}
 
-		item, err := ks.db.Get(key)
-		if err != nil {
-			return false, fmt.Errorf("Get error, err - %s", err)
-		}
-
-		if len(item.Data) == 0 {
-			// return sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key)
-			return false, fmt.Errorf("sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key), err: %s", sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key))
-		}
-
-		// 2.try to deserialize using proto, if good then continue, otherwise try to deserialize using amino
-		if _, err := ks.protoUnmarshalRecord(item.Data); err == nil {
-			fmt.Println("protoUnmarshalRecord continue")
-			//TODO decide if we do not migrate, we should mark it as false or true?
-			migrations[i] = false 
-			continue
-		}
-
-		legacyInfo, err := unMarshalInfo(item.Data)
-		if err != nil {
-			unmarshalErr := fmt.Errorf("unable to unmarshal item.Data, err: %w", err)
-			fmt.Println(unmarshalErr)
-			migrations[i] = false
-			continue
-		}
-		//4.serialize info using proto
-		k, err := ks.convertFromLegacyInfo(legacyInfo)
+		migrated, err = ks.migrate(key)
```

Comment:
I run `migrate` func for every key. If `migrate` outputs an error I exit `CheckMigrate`, otherwise for loop continues. Does it sound good to you? cc @robert-zaremba  @amaurym

Reply:
As discussed on Discord, we should have two execution branches:

1. When accessing a single key (eg when doing a tx using CLI) we should unmarshal proto, and if that fails, we should check for migration (unmarshal amino - and if this succeeds -> migrate that single key).
2. When running `app keys migrate` will check if we should migrate and migrate all keys in the loop. If one of the keys will fail we will continue.